### PR TITLE
Fixed expiration url time

### PR DESCRIPTION
**Pull request by dapariciom**
Fixed expiration url time

### CODE REVIEW

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java [hunk 1]
```diff
@@ -60,27 +59,25 @@ public ResponseEntity<UrlResponse> shortUrl(@RequestBody UrlRequest urlRequest)
...
-            ResponseEntity.status(HttpStatus.BAD_REQUEST).headers(headers).build();
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Url not found");
         }
 
-        //TODO: Handle url is has expired
-//        if(url.getExpirationDate().isBefore(LocalDateTime.now())){
-//            ResponseEntity.status(HttpStatus.BAD_REQUEST).headers(headers).build();
-//        }
+        if(url.getExpirationDate().isBefore(LocalDateTime.now())){
+            throw new ResponseStatusException(HttpStatus.OK, "Url has expired");
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Let's go with 410 Status Code Error here

File: src/main/java/com/example/shorturl/service/IShortenUrlService.java [hunk 2]
```diff
@@ -16,4 +16,15 @@ public interface IShortenUrlService {
...
+
+
+    /**
+     * This method uses an encoded short URL to obtain a linked long URL
+     *
+     * @param url This is the short URL in String format
+     * @return UrlEntity This returns the entity that contains the long URL and
+     * other related parameters
+     */
+    UrlEntity getEncodedUrl(String url);
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/service/IShortenUrlService.java]: Optional

File: src/main/java/com/example/shorturl/service/ShortenUrlService.java [hunk 3]
```diff
@@ -23,6 +23,8 @@
 @Service
 public class ShortenUrlService implements IShortenUrlService{
 
+    long EXPIRATION_TIME_HOURS = 1;
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/service/ShortenUrlService.java]: Let's have 20s for now

Comment (BeeAM17) [reply src/main/java/com/example/shorturl/service/ShortenUrlService.java]: You can use time units library or smth

File: src/main/java/com/example/shorturl/service/ShortenUrlService.java [hunk 4]
```diff
@@ -67,7 +66,7 @@ public UrlEntity getEncodedUrl(String url){
         return  urlList.get(0);
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/service/ShortenUrlService.java]: Let's think about better approach and save in this method itself

**Review by BeeAM17 — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/shorturl/service/ShortenUrlService.java (hunk 3 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/service/ShortenUrlService.java]: Fixed

File: src/main/java/com/example/shorturl/service/ShortenUrlService.java (hunk 4 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/service/ShortenUrlService.java]: Checked

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java (hunk 1 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Fixed

File: src/main/java/com/example/shorturl/service/IShortenUrlService.java (hunk 2 above)
Comment (dapariciom) [reply src/main/java/com/example/shorturl/service/IShortenUrlService.java]: Fixed

File: src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java [hunk 5]
```diff
@@ -60,27 +59,27 @@ public ResponseEntity<UrlResponse> shortUrl(@RequestBody UrlRequest urlRequest)
...
         }
 
-        //TODO: Handle url is has expired
-//        if(url.getExpirationDate().isBefore(LocalDateTime.now())){
-//            ResponseEntity.status(HttpStatus.BAD_REQUEST).headers(headers).build();
-//        }
+        UrlEntity url = optionalUrl.get();
+
+        if(url.getExpirationDate().isBefore(LocalDateTime.now())){
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Url has expired");
```
Comment (BeeAM17) [inline src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: 410, not 401 :)

Comment (dapariciom) [reply src/main/java/com/example/shorturl/controller/api/ShortenUrlController.java]: Sorry hehehe fixing it

**Review by BeeAM17 — APPROVED** [review]
Nice work

**Pull request merged**